crates/re_renderer/src/renderer/depth_cloud.rs

```diff
@@ -242,12 +242,28 @@ fn create_and_upload_texture<T: bytemuck::Pod>(
     let width_blocks = depth_cloud.depth_dimensions.x / format_info.block_dimensions.0 as u32;
     let bytes_per_row_unaligned = width_blocks * format_info.block_size as u32;
 
+    // TODO(andreas): CpuGpuWriteBelt should make it easier to do this.
+    let bytes_per_row_aligned =
+        wgpu::util::align_to(bytes_per_row_unaligned, wgpu::COPY_BYTES_PER_ROW_ALIGNMENT);
+
     let mut depth_texture_staging = ctx.cpu_write_gpu_read_belt.lock().allocate::<T>(
         &ctx.device,
         &ctx.gpu_resources.buffers,
         data.len(),
     );
-    depth_texture_staging.extend_from_slice(data);
+
+    // Fill with a single copy if possible, otherwise do multiple, filling in padding.
+    if bytes_per_row_aligned == bytes_per_row_unaligned {
+        depth_texture_staging.extend_from_slice(data);
+    } else {
+        for row in data.chunks(width_blocks as usize) {
+            depth_texture_staging.extend_from_slice(row);
+            depth_texture_staging.extend(std::iter::repeat(T::zeroed()).take(
```

Comment:
it would be faster to crate a slice here ahead of time and copy that in several times